### PR TITLE
[main] Bump go toolchain to 1.22.7

### PR DESCRIPTION
**Pull request by henrybear327**
Reference:
- https://github.com/etcd-io/etcd/issues/18548
~~- https://github.com/etcd-io/etcd/issues/18443~~

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Comment by henrybear327** [comment]
@ivanvc The devcontainer can't be updated now, because at the time of writing the 1.23 image hasn't released yet.

Please see: https://hubgw.docker.com/r/microsoft/devcontainers-go

**Comment by henrybear327** [comment]
@ivanvc  I will resolve the CI error in the meantime. But we probably would be blocked by https://github.com/etcd-io/etcd/pull/18444#issuecomment-2289978906 in any case.

**Comment by ahrtr** [comment]
For the `pull-etcd-govulncheck` failure, it should be caused the by environment. We might want to raise an issue to the kubernetes/test-infra

**Comment by jmhbnz** [comment]
Hey Team - The devcontainer is now available following https://github.com/devcontainers/images/pull/1167

**Comment by henrybear327** [comment]
> For the `pull-etcd-govulncheck` failure, it should be caused the by environment. We might want to raise an issue to the kubernetes/test-infra

Hello @ahrtr, 

Thanks for the idea and insight!

 May I ask where exactly I should raise this issue?

Thank you!

**Comment by ivanvc** [comment]
> May I ask where exactly I should raise this issue?

I think https://github.com/kubernetes/test-infra, the image that may have the issue is: https://github.com/kubernetes/test-infra/tree/master/images/kubekins-e2e-v2.

**Comment by henrybear327** [comment]
> > May I ask where exactly I should raise this issue?
> 
> I think https://github.com/kubernetes/test-infra, the image that may have the issue is: https://github.com/kubernetes/test-infra/tree/master/images/kubekins-e2e-v2.

Thanks @ivanvc for the hint. 

I dug into the issue. As the k/k is still on 1.22, thus [all the images](https://github.com/kubernetes/test-infra/blob/master/images/kubekins-e2e-v2/variants.yaml) are on 1.22 and not 1.23.

I am thinking of submitting a PR to add a new variant that contains 1.23, but not sure if this is the best way to do it.

@ivanvc do you have some ideas? I am happy to get involved in the prow / test-infra to get into this new field! :)

**Comment by ivanvc** [comment]
@henrybear327, if you have the time and know what to do, I'd say go for it. :)

**Comment by ivanvc** [comment]
@henrybear327, 1.23.1 was released today. Could you please update this PR to bump to the latest patch? Thanks.

**Comment by henrybear327** [comment]
> @henrybear327, 1.23.1 was released today. Could you please update this PR to bump to the latest patch? Thanks.


Ok! 

Should I bump others too?

**Comment by ivanvc** [comment]
Yes, if you want, please refer to #18548 (and assign it to you). I'm doing 3.4 and 3.5 because we want to expedite them for the releases.

File: .go-version [hunk 1]
```diff
@@ -1 +1 @@
-1.22.6
+1.23.0
```
Comment (ahrtr) [inline .go-version:1]:
```suggestion
1.23.1
```

**Comment by ivanvc** [comment]
/retitle [main] Bump go toolchain to 1.23.1

**Comment by henrybear327** [comment]
@ahrtr @ivanvc as we have discussed at https://github.com/etcd-io/etcd/issues/18548#issuecomment-2343061825 and offline, we will use 1.22.7 for the main branch now. 

I will work on the `test-infra` image in the meantime and if I can get the 1.23.1 image working we can then upgrade the main branch to 1.23.1!

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks

**Review by jmhbnz — APPROVED** [review]
LGTM - Thanks @henrybear327

**Pull request merged**